src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -0,0 +1,825 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Dataflow
+{
+	/// <summary>
+	/// Tracks information about the contents of a stack slot
+	/// </summary>
+	class StackSlot
+	{
+		public ValueNode Value { get; set; }
+
+		/// <summary>
+		/// True if the value is on the stack as a byref
+		/// </summary>
+		public bool IsByRef { get; set; }
+
+		public StackSlot ()
```

Comment:
Could this be a private and used as static instance value?

Reply:
We'll get bigger bang for the buck if we make `StackSlot` a `struct`. I'm not clear on why it was a class in .NET Native.

We might want to also explore getting rid of `StackSlot` completely and just use a stack of `ValueNodes`. The `IsByRef` tracking might not be necessary. .NET Native also tracked the static type of the value in the slot but we don't have that here because this analysis doesn't care.

Reply:
I'll make an issue with a running list of these things. Resolving them all would be a necessary prerequisite for merging to master.